controllers/session.go

```diff
@@ -0,0 +1,69 @@
+package controllers
+
+import (
+	"net/http"
+
+	"go-google-scraper-challenge/forms"
+
+	"github.com/beego/beego/v2/server/web"
+)
+
+// SessionController operations for User
+type SessionController struct {
+	BaseController
+}
+
+// URLMapping map user controller actions to functions
+func (c *SessionController) URLMapping() {
+	c.Mapping("New", c.New)
+	c.Mapping("Create", c.Create)
+}
+
+// New handle new session action
+// @Title New
+// @Description new session
+// @Success 200
+// @router / [get]
+func (c *SessionController) New() {
+	c.EnsureGuestUser()
+
+	c.Data["Title"] = "Sign In"
+
+	c.Layout = "layouts/authentication.tpl"
+	c.TplName = "sessions/new.tpl"
+
+	web.ReadFromRequest(&c.Controller)
+}
+
+// Create handle create session action
+// @Title Create
+// @Description create session
+// @Success 302 redirect to root path with success message
+// @Failure 302 redirect to login path with error message
+// @router / [post]
+func (c *SessionController) Create() {
```

Comment:
Shouldnt this method also calls `c.EnsureGuestUser()` to guarantee that only non-authenticated users can login? Since it's a POST request, it would require to handle the response differently e.g. it could be 403 in the sense that the data sent is correct (so not 400 - bad request) and it's just the operation is not allowed if the user is already logged in.

Reply:
@olivierobert Updated on [281d26a](https://github.com/carryall/go-google-scraper-challenge/pull/31/commits/281d26ab4969ba60f87021ae92b88f0202db4f89) and also add test cases when user already signed in on [38f9369](https://github.com/carryall/go-google-scraper-challenge/pull/31/commits/38f9369ef670ae56ec0325222792588c8f4ff5f5)